source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRMediaRenderer.mm

```diff
@@ -43,26 +43,37 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
     std::shared_ptr<Media> mediaElem = std::dynamic_pointer_cast<Media>(elem);
 
     NSMutableDictionary *imageViewMap = [rootView getImageMap];
-    NSString *key = [NSString stringWithCString:mediaElem->GetPoster().c_str() encoding:[NSString defaultCStringEncoding]];
-    UIImage *img = imageViewMap[key];
+    NSString *urlString = [NSString stringWithCString:mediaElem->GetPoster().c_str() encoding:[NSString defaultCStringEncoding]];
+    NSString *numberString = [[NSNumber numberWithUnsignedLongLong:(unsigned long long)(elem.get())] stringValue];
+    NSString *piikey = [NSString stringWithCString:[acoConfig getHostConfig] -> GetMedia().playButton.c_str() encoding:[NSString defaultCStringEncoding]];
+    NSString *piikeyViewIF = [NSString stringWithFormat:@"%llu_playIcon", (unsigned long long)elem.get()];
+
+    NSDictionary *pieces = @{
+        @"number" : numberString,
+        @"url" : urlString,
+        @"playicon-url-imageView" : piikey,
+        @"playicon-url-viewIF" : piikeyViewIF
+    };
+
+    NSString *mediaKey = makeKeyForImage(acoConfig, @"media-poster", pieces);
+    UIImage *img = imageViewMap[mediaKey];
     UIImageView *view = nil;
-    CGFloat heightToWidthRatio = 0.0f;
     ACRContentHoldingUIView *contentholdingview = nil;
 
     // if poster is available, restrict the image size to the width of superview, and adjust the height accordingly
     if (img) {
-        view = [[UIImageView alloc] initWithImage:img];
-
-        if (img.size.width > 0) {
-            heightToWidthRatio = img.size.height / img.size.width;
+        contentholdingview = (ACRContentHoldingUIView *)[rootView getImageView:mediaKey];
```

Comment:
>(ACRContentHoldingUIView *)[ [](start = 29, length = 28)

Curious: Is there something like a safe casting in ObjectiveC? Do we have the right level of compiler flags enabled to catch potential issues?

Reply:
Objective-C is a superset of c, but there are no other ways of casting.